### PR TITLE
added disclaimer about network segments due to Serf limitations

### PR DESCRIPTION
**Pull request by thepomeranian**

### CODE REVIEW

File: website/source/docs/enterprise/network-segments/index.html.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ with each other.
 To get started with Network Segments,
 [read the guide](https://learn.hashicorp.com/consul/day-2-operations/network-segments).
 
+-> **Note:** Due to limitations in Consul's implementation of Serf, you may currently only use 10 Network Segments per Consul cluster.
```
Comment (freddygv) [inline website/source/docs/enterprise/network-segments/index.html.md]:
```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you may currently only use up to 10 Network Segments per Consul cluster.
```

File: website/source/docs/enterprise/network-segments/index.html.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ with each other.
 To get started with Network Segments,
 [read the guide](https://learn.hashicorp.com/consul/day-2-operations/network-segments).
 
+-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you may currently only use up to 10 Network Segments per Consul cluster.
```
Comment (kaitlincarter-hc) [inline website/source/docs/enterprise/network-segments/index.html.md]:
```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), you can only use up to 10 Network Segments per Consul cluster.
```
Did we decide if a Serf version number should be added?

Comment (tryan225) [inline website/source/docs/enterprise/network-segments/index.html.md]:
```suggestion
-> **Note:** Due to limitations in [Serf](https://www.consul.io/docs/internals/gossip.html), a Consul agent configured with too many Network Segments may not be able to start```

Comment (tryan225) [reply website/source/docs/enterprise/network-segments/index.html.md]: We should probably remove the arbitrary value of 10 Network Segments.

Comment (judithpatudith) [reply website/source/docs/enterprise/network-segments/index.html.md]: +1 if 10 isn't a real number we shouldn't say it, or we should hedge with "as few as" or something?

**Comment by danielehc** [comment]
Closing as the change is being made in https://github.com/hashicorp/consul/pull/7004

**Pull request closed without merging**